frontend/src/components/RepositoryPanel.vue

```diff
@@ -468,10 +472,18 @@ export default defineComponent({
 
       const disableSQLReview = disableSQLReviewCI.value;
 
-      await repositoryStore.updateRepositoryByProjectId({
-        projectId: props.project.id,
-        repositoryPatch,
-      });
+      try {
+        await repositoryStore.updateRepositoryByProjectId({
+          projectId: props.project.id,
+          repositoryPatch,
+        });
+      } catch (error) {
+        // Branch not found.
+        if ((error as AxiosError)?.response?.status === 404) {
```

Comment:
> the 404 code is too general

Agree. The not found error can be either branch not found or vcs/project not found. It is hard for the client side to know the exact error. But 400 (`http.StatusBadRequest`) is a general choice, too.

https://github.com/bytebase/bytebase/blob/24bc08b57790def0d673de34739ad16e842064c4/server/project.go#L232-L246

I think the best way may be to use customized status codes so that we can accurately tell from the client side which error is the underlying one. But this may take much more effort.

In my opinion, it's OK to just show messages in the notification popup. @d-bytebase @candy2255 WDYT?

Reply:
> the 404 code is too general

Agree. Similar things about the 401 code happen. See BYT-1241

Reply:
OK. It would be great to make the message easy to read